src/scancode/resource.py

```diff
@@ -120,6 +120,39 @@ class ResourceNotInCache(Exception):
 class UnknownResource(Exception):
     pass
 
+def depth_walk(root_location, max_depth, skip_ignored=lambda x: False, error_handler=lambda:None):
+    """ Generator that walks the `root_location` directory recursively upto `max_depth` of 
+    nested subdirectories and yields a (top, dirs, files) tuple at each step. 
+
+    Arguments: 
+       - root_location: Absolute path for the directory to be walked
+       - max_depth: Positive integer for fixed depth limit. negative for no limit
+       - skip_ignored: Function that returns boolean indicating whether to ignore searching files
```

Comment:
What about instead moving the err and is_ignored functions here from https://github.com/nexB/scancode-toolkit/pull/1948/files#diff-a8e64273dc1cb3ca2a1ff8127a989c6cR433 ... that way you do not need these args anymore?

Reply:
We can do that for `skip_ignored` but `err` has to be bound to `Codebase` as it updates the `errors` attribute on the object. The idea of having them as option callback arguments was to allow some flexibility in case the functionality is reused somewhere else. What do you think?

Reply:
OK, your call... but consider also YAGNI https://en.wikipedia.org/wiki/You_aren't_gonna_need_it ... it is easy to add if really needed, hard to remove otherwise.

Reply:
Yeah, that is right. I have taken the `skip_ignored` function out, so we have one less argument. The same couldn't be done in a neat way for `err`.

Reply:
Yeah, you're right. I have taken `skip_ignored` out, so we have one less argument.